sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -368,16 +384,27 @@ private Tree createBinaryExpression(TreeMetaData metaData, KtBinaryExpression el
       return createNativeTree(metaData, new KotlinNativeKind(element, operationToken), children);
     }
 
-    Operator operator = TOKENS_OPERATOR_MAP.get(operationToken);
+    Operator operator = BINARY_OPERATOR_MAP.get(operationToken);
     AssignmentExpressionTree.Operator assignmentOperator = ASSIGNMENTS_OPERATOR_MAP.get(operationToken);
     if (operator != null) {
       return new BinaryExpressionTreeImpl(metaData, operator, leftOperand, rightOperand);
     } else if (assignmentOperator != null) {
       return new AssignmentExpressionTreeImpl(metaData, assignmentOperator, leftOperand, rightOperand);
     } else {
       // FIXME ensure they are all supported. Ex: Add '/=' for assignments
-      return createOperationExpression(metaData, element);
+      return createNativeOperationExpression(metaData, element);
+    }
+  }
+
+  private Tree createUnaryExpression(TreeMetaData metaData, KtUnaryExpression element) {
+    Tree operand = createElement(element.getBaseExpression());
+    IElementType operationToken = element.getOperationToken();
```

Comment:
Why is the way we get the operationToken different in the unary and binary expressions?

Reply:
For some reason `getOperationToken()` do not return the same object when called on `KtUnaryExpression ` or `KtBinaryExpression`.
Binary espression returns a `KtOperationReferenceExpression`, which has the `getOperationSignTokenType()` method returning a typed `KtToken`.
But unary expression `getOperationToken()` returns a `KtSimpleNameExpression` instead. So I decided to use the `getOperationToken();` method of the expression.
Basically, it is possible to get a `KtToken` type directly from binary expression, but not in unary expression.

Reply:
Do we want to change the BinaryExpression method to follow the same logic as UnaryExpression for consistency?

Reply:
I'd prefer to keep it as it is as it has the type safety to be a `KtToken` for binary epxression.
I'd have liked the same thing for unary expression, but it is not available for some reason